fetchai/ledger/transaction.py

```diff
@@ -114,6 +121,59 @@ def data(self):
     def data(self, data: bytes):
         self._data = bytes(data)
 
+    def compare(self, other: 'Transaction') -> bool:
+        if not self.from_address == other.from_address:
+            return False
+        if not self.transfers == other.transfers:
+            return False
+        if not self.valid_from == other.valid_from:
+            return False
+        if not self.valid_until == other.valid_until:
+            return False
+        if not self.charge_rate == other.charge_rate:
+            return False
+        if not self.charge_limit == other.charge_limit:
+            return False
+        if not self.contract_digest == other.contract_digest:
+            return False
+        if not self.contract_address == other.contract_address:
+            return False
+        if not self.chain_code == other.chain_code:
+            return False
+        if not self.action == other.action:
+            return False
+        if not self.shard_mask == other.shard_mask:
+            return False
+        if not self.data == other.data:
+            return False
+        if not self.signers.keys() == other.signers.keys():
+            return False
+        if not self.counter == other.counter:
+            return False
+        if not self._metadata == other._metadata:
+            return False
+        if not self.payload == other.payload:
```

Comment:
This needs refactoring since payload is only property that needs comparing, others all constitute payload and don't need comparing individually.